source/__tests__/integration/new-profile.test.js

```diff
@@ -41,19 +41,32 @@ describe('Adding and removing a new Profile', () => {
         await expect(popover).toBe('Link to more information about profile')
       })
 
-      it('validates with checkURL', async () => {
+      it('with function checkURL', async () => {
         const ng_blur = await page.$eval(source_input_sel, e => e.getAttribute('ng-blur'))
         await expect(ng_blur).toBe('checkURL()')
       })
+
+      it('displays error if source and user switches focus', async () => {
+        await page
+          .type(source_input_sel, '')
+          .catch(error => console.log(`promise error for : ${error}`))
+        await page.keyboard.press(`Tab`)
+        await expect_value_in_selector_textContent('#alertBox > div > div > div.modal-header > h3', 'Error!')
+        const invalid_url_class = page.$('input#resourceURI', e => e.getAttribute('ng-invalid-url'))
+        expect(invalid_url_class).toBeDefined()
+        await page.$eval("button#alertClose", e => e.click())
+        await page.waitFor(1000)
+      })
+
     })
 
     describe('Adding a Resource Template', () => {
 
-        beforeAll(async () => {
-            await page.waitForSelector('#addResource')
-            await page.click('#addResource')
-            await page.waitFor(1000)
-        })
+      beforeAll(async () => {
+          await page.waitForSelector('#addResource')
+          await page.click('#addResource')
+          await page.waitFor(1000)
```

Comment:
is there a way this waitFor() can be made conditional?

Reply:
Updated #137 to point to this use of `waitFor`.